cmd/rqlite/main.go

```diff
@@ -127,7 +129,20 @@ func sendRequest(ctx *cli.Context, urlStr string, line string, argv *argT, ret i
 
 	nRedirect := 0
 	for {
-		resp, err := client.Post(url, "application/json", strings.NewReader(data))
+		req, err := http.NewRequest("POST", url, strings.NewReader(data))
+		if err != nil {
+			return err
+		}
+
+		if argv.Credentials != "" {
+			user, pass, err := net.SplitHostPort(argv.Credentials)
```

Comment:
While this works, `SplitHostPort` is not for splitting Basic Auth credentials. Instead please do:
```
creds := strings.Split(arg, ":")`
if len(creds) != 2 || creds[0] == "" || creds[1] == "" {
    return fmt.Errorf("invalid Basic Auth credentials format")
}
req.SetBasicAuth(creds[0], creds[1])
```
or something to this effect.

Reply:
Ok, that's fine. However, do you want to mandate a non-empty username and password? Because RFC 2617 doesn't. I'll make the change later today. Thanks.